examples/c/ex_col_store.c

```diff
@@ -0,0 +1,585 @@
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *result = temp;
+
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        if (temp > *result) {
+            *result = temp;
+        }
+
+        /* For debugging purposes */
+        printf("ID %" PRIu64, recno);
+        printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+    }
+
+    return true;
+}
+/*! [Obtains the average data across all fields given a specific location] */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    const char *day;
+    const char *country;
+    uint64_t recno;
+    uint16_t hour, pressure, loc_long, loc_lat;
+    uint8_t temp, humidity, wind, feels_like_temp;
+    /* num_rec is the total number of records we're obtaining averages for. */
+    int ret, num_rec = 5, exact;
+    unsigned int count = 0;
+    /* rec_arr holds the sum of the records in order to obtain the averages. */
+    unsigned int rec_arr[5] = {0, 0, 0, 0, 0};
+    /* Open a cursor to search for the location, currently RUS. */
+    error_check(
+      session->open_cursor(session, "index:weathertable:country", NULL, NULL, &loc_cursor));
+    loc_cursor->set_key(loc_cursor, "RUS\0\0");
+    error_check(loc_cursor->search_near(loc_cursor, &exact));
+    /* Error handling in the case RUS is not found. In this case as it's a hardcoded location,
+     *  if there aren't any matching locations, no average data is obtained.
+     * */
+    if (exact != 0) {
+        return;
+    }
+    /* Populate the array with the totals of each of the columns. */
+    while ((ret = loc_cursor->next(loc_cursor)) == 0) {
+        count++;
+        error_check(loc_cursor->get_key(loc_cursor, &recno));
+        error_check(loc_cursor->get_value(loc_cursor, &day, &hour, &temp, &humidity, &pressure,
+          &wind, &feels_like_temp, &loc_lat, &loc_long, &country));
+        /* Increment the values of the rec_arr with the temp_arr values. */
+        rec_arr[0] += temp;
+        rec_arr[1] += humidity;
+        rec_arr[2] += pressure;
+        rec_arr[3] += wind;
+        rec_arr[4] += feels_like_temp;
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(loc_cursor->close(loc_cursor));
+    /* For debugging purposes. */
+    printf("Number of matching entries: %u \n", count);
+    /* Get the average values by dividing with the total number of records. */
+    for (int i = 0; i < num_rec; i++) {
+        rec_arr[i] = rec_arr[i] / count;
+    }
+    /* List the average records */
+    printf("Average records for location RUS : \nTemp: %" PRIu8 ", Humidity: %" PRIu8
+           ", Pressure: %" PRIu16 ", Wind: %" PRIu8 ", Feels like: %" PRIu8 "\n",
+      rec_arr[0], rec_arr[1], rec_arr[2], rec_arr[3], rec_arr[4]);
+}
+
+int
+main(int argc, char *argv[])
+{
+    WEATHER *w;
```

Comment:
There's a convention on how these declarations are ordered.

It should be something like:

1. Structs in alphabetical order of the type name.
2. Typedefs (like `wt_timestamp`) in alphabetical order.
3. Primitives (`char`, `int`, etc) or standard typedefs (`uint32_t`) in order from largest to smallest memory size.

That's just off the top of my head. Take a look at some other source files to confirm.